packages/mapviewer/src/modules/menu/components/advancedTools/ImportCatalogue/useProviders.js

```diff
@@ -8,6 +9,16 @@ class Provider {
         this.htmlDisplay = url.replace(/access_key=.*/, '')
         this.emphasize = emphasize
     }
+
+    getBaseUrl() {
+        try {
+            const urlObj = new URL(this.url)
+            return urlObj.origin
+        } catch (e) {
+            log.error('Invalid URL:', this.url, e)
+            return ''
```

Comment:
might make more sense to return `null` or `undefined` and then manage the fact that it can be "falsy" in `useProviders` function bellow